src/Adaptivity/AdaptedTriangulations.jl

```diff
@@ -0,0 +1,316 @@
+end
+
+# Wrap constructors
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::AdaptedDiscreteModel,filter::AbstractArray) where d
+  
+  trian = Triangulation(ReferenceFE{d},get_model(model),filter)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::AdaptedDiscreteModel,labels::FaceLabeling;kwargs...) where d
+  trian = Triangulation(ReferenceFE{d},get_model(model),labels;kwargs...)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(trian::AdaptedTriangulation,args...;kwargs...)
+  return AdaptedTriangulation(Triangulation(trian.trian,args...;kwargs...),trian.adapted_model)
+end
+
+function Geometry.BoundaryTriangulation(model::AdaptedDiscreteModel,args...;kwargs...)
+  trian = BoundaryTriangulation(get_model(model),args...;kwargs...)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.SkeletonTriangulation(model::AdaptedDiscreteModel,face_to_mask::AbstractVector{Bool})
+  trian = SkeletonTriangulation(get_model(model),face_to_mask)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.InterfaceTriangulation(model::AdaptedDiscreteModel, cell_to_is_in::Vector{Bool})
+  trian = InterfaceTriangulation(get_model(model),cell_to_is_in)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.InterfaceTriangulation(model::AdaptedDiscreteModel,cell_to_inout::AbstractVector{<:Integer})
+  trian = InterfaceTriangulation(get_model(model),cell_to_inout)
+  return AdaptedTriangulation(trian,model)
+end
+
+function Geometry.is_change_possible(strian::AdaptedTriangulation,ttrian::AdaptedTriangulation)
+  # A) Both Triangulations are exactly the same
+  (strian === ttrian) && (return true)
+
+  # B) Same background model -> Default change of Triangulation
+  if (get_background_model(strian) === get_background_model(ttrian))
+    return is_change_possible(strian.trian,ttrian.trian)
+  end
+
+  # C) Different background model, but same type of Triangulation (Skeleton, BodyFitted, View, ...)
+  if typeof(strian.trian) == typeof(ttrian.trian) # Is this too restrictive???
+    return is_related(strian,ttrian)
+  end
+
+  # D) Different background model AND different type of triangulation
+  @notimplemented
+  return false
+end
+
+function Geometry.is_change_possible(strian::AdaptedTriangulation,ttrian::Triangulation)
+  # A) Same background model -> Default change of Triangulation
+  if (get_background_model(strian) === get_background_model(ttrian))
+    return is_change_possible(strian.trian,ttrian)
+  end
+  
+  # B) Different background model, but same type of Triangulation (Skeleton, BodyFitted, View, ...)
+  if typeof(strian.trian) == typeof(ttrian)
+    return is_related(strian,ttrian)
+  end
+
+  # C) Different background model AND different type of triangulation
+  @notimplemented
+  return false
+end
+
+function Geometry.is_change_possible(strian::Triangulation,ttrian::AdaptedTriangulation)
```

Comment:
Might we use here meta-programming to avoid "replicating" code?

Reply:
Addressed in [1d41637](https://github.com/gridap/Gridap.jl/pull/838/commits/1d41637e9b9eee0bce56b6b7ba99c01e435c4eea)